cmake/extensions.cmake

```diff
@@ -2110,6 +2110,40 @@ function(zephyr_check_cache variable)
   set(CACHED_${variable} ${${variable}} CACHE STRING "Selected ${variable_text}")
 endfunction(zephyr_check_cache variable)
 
+
+# Usage:
+#   zephyr_boilerplate_watch(SOME_BOILERPLATE_VAR)
+#
+# Inform the build system that SOME_BOILERPLATE_VAR, a variable
+# handled in cmake/app/boilerplate.cmake, is now fixed and should no
+# longer be changed.
+#
+# This function uses variable_watch() to print a noisy warning
+# if the variable is set after it returns.
+function(zephyr_boilerplate_watch variable)
+  variable_watch(${variable} zephyr_variable_set_too_late)
+endfunction()
+
+function(zephyr_variable_set_too_late variable access value current_list_file)
```

Comment:
How about something like this as function name:
`zephyr_variable_warning` or `zephyr_access_warning` ?

Reply:
Discussed offline: we'll keep this as-is since it makes the call stacks more readable.